.github/scripts/determine-initial-guest-os-versions.py

```diff
@@ -0,0 +1,148 @@
+import json
+from typing_extensions import Any
+from urllib.request import Request, urlopen
+import urllib.error
+import sys
+
+ROLLOUT_DASHBOARD_ENDPOINT='https://rollout-dashboard.ch1-rel1.dfinity.network/api/v1/rollouts'
+PUBLIC_DASHBOARD_ENDPOINT='https://ic-api.internetcomputer.org/api/v3/subnets?format=json'
+
+# Key definitions
+STATE = 'state'
+FAILED = 'failed'
+COMPLETE = 'complete'
+BATCHES = 'batches'
+SUBNETS = 'subnets'
+GIT_REVISION = 'git_revision'
+EXECUTED_TIMESTAMP_SECONDS = 'executed_timestamp_seconds'
+REPLICA_VERSIONS = 'replica_versions'
+REPLICA_VERSION_ID = 'replica_version_id'
+
+def eprint(*args, **kwargs):
+    print(*args, file=sys.stderr, **kwargs)
+
+def maybe_fetch_data(url : str, timeout : float) -> str | None:
+    """Try fetch data from `url`.
+
+    Will try to fetch data and return `None` if the resource is unavailable for some.
+    If this call fails other resources should be considered.
+    """
+    try:
+        req = Request(url)
+        req.add_header('accept', '*/*')
+        req.add_header('user-agent', 'dfinity-ci')
+        with urlopen(req, timeout=timeout) as response:
```

Comment:
Why not use `request.get()` from the requests library, and then `r.raise_for_status()` to handle non-200 responses?  urlopen?

Reply:
This is native python, no need to waste time on installing pip requirements. Even the repo is sparse-checked out for only that file.